ambari-metrics-timelineservice/src/main/java/org/apache/ambari/metrics/core/timeline/source/cache/InternalMetricsCache.java

```diff
@@ -181,49 +161,14 @@ public void putAll(Collection<TimelineMetrics> metrics) {
     }
   }
 
-  class InternalCacheEvictionListener implements CacheEventListener {
-
-    @Override
-    public void notifyElementRemoved(Ehcache cache, Element element) throws CacheException {
-      // expected
-    }
-
-    @Override
-    public void notifyElementPut(Ehcache cache, Element element) throws CacheException {
-      // do nothing
-    }
-
+  class InternalCacheEvictionListener implements CacheEventListener<InternalMetricCacheKey, InternalMetricCacheValue> {
     @Override
-    public void notifyElementUpdated(Ehcache cache, Element element) throws CacheException {
-      // do nothing
-    }
-
-    @Override
-    public void notifyElementExpired(Ehcache cache, Element element) {
-      // do nothing
-    }
-
-    @Override
-    public void notifyElementEvicted(Ehcache cache, Element element) {
-      // Bad - Remote endpoint cannot keep up resulting in flooding
-      InternalMetricCacheKey key = (InternalMetricCacheKey) element.getObjectKey();
-      LOG.warn("Evicting element from internal metrics cache, metric => " + key
-        .getMetricName() + ", startTime = " + new Date(key.getStartTime()));
-    }
-
-    @Override
-    public void notifyRemoveAll(Ehcache cache) {
-      // expected
-    }
-
-    @Override
-    public Object clone() throws CloneNotSupportedException {
-      return null;
-    }
-
-    @Override
-    public void dispose() {
-      // do nothing
+    public void onEvent(CacheEvent<? extends InternalMetricCacheKey, ? extends InternalMetricCacheValue> event) {
+      if (event.getType() == EventType.EVICTED) {
+        InternalMetricCacheKey key = event.getKey();
+        LOG.warn("Evicting element from internal metrics cache, metric => " + key
```

Comment:
nit: can we use {} placeholders?

Reply:
@virajjasani Apache Commons Logging does not support {} placeholders directly.